client/extensions/woocommerce/app/products/product-create.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentlyEditingProduct } from '../../state/ui/products/selectors';
+import { editProduct, editProductAttribute } from '../../state/ui/products/actions';
+import ProductForm from './product-form';
+
+class ProductCreate extends Component {
+
+	componentDidMount() {
+		const { product } = this.props;
+
+		if ( ! product ) {
+			this.props.editProduct( null, {
+				type: 'simple'
+			} );
+		}
+	}
+
+	componentWillUnmount() {
+		// TODO: Remove the product we added here from the edit state.
+	}
+
+	render() {
+		const { product } = this.props;
+
+		return (
+			<ProductForm
+				product={ product || {} }
```

Comment:
Yeah, that's a fair point. Maybe the best thing to do is not render ProductForm until we have a valid product? In this case, the user would never see it without as it's added upon mount.